Makefile

```diff
@@ -0,0 +1,39 @@
+NAME    := munge
+SRC_EXT := rpm
+TARBALL_VERSION = 0.5.13
+VERSION := $(TARBALL_VERSION)-lp151.3.3
+URL_BASE := https://download.opensuse.org/source/distribution/leap/15.1
+URL_BASE := $(URL_BASE)/repo/oss/src
+SOURCE   = $(URL_BASE)/$(NAME)-$(VERSION).src.$(SRC_EXT)
+
+SPEC := _topdir/SOURCES/munge.spec
+
+RPMS := _topdir/RPMS/x86_64/lib$(NAME)2-$(VERSION).x86_64.rpm
+RPMS += _topdir/RPMS/x86_64/$(NAME)-devel-$(VERSION).x86_64.rpm
+RPMS += _topdir/RPMS/x86_64/$(NAME)-$(VERSION).x86_64.rpm
+SRPM := _topdir/SRPMS/$(NAME)-$(VERSION).src.$(SRC_EXT)
+
+SOURCES := _topdir/SOURCES/$(NAME)-$(TARBALL_VERSION).tar.gz
+
+COMMON_RPM_ARGS := --define "%_topdir $$PWD/_topdir"
+SRPM_IN := _topdir/SOURCES/$(NAME)-$(VERSION).src.$(SRC_EXT)
+
+# Can not just CURL the file into the _topdir/SRPMS, the rpmbuild of the
+# binary packages replaces it, which makes it newer than the unpacked
+# sources.
+$(SRPM_IN): Makefile
+	rm -rf _topdir
+	mkdir -p _topdir/SOURCES
+	cd _topdir/SOURCES/; curl -f -L -O '$(SOURCE)'
+
+$(SRPM): $(SPEC)
+	mkdir -p _topdir/SRPMS
+	ln -f $< $@
+
+$(SOURCES): $(SRPM_IN)
+	cd _topdir/SOURCES; rpm2cpio ../../$(SRPM_IN) | cpio -iv
+
+$(SPEC): $(SOURCES)
```

Comment:
Instead of all of this, why not just import the specfile from the RPM that you are backporting so that this repo operates just like any other RPM-building repo?

Reply:
You fetch the SRPM that you want to backport, unpack it and commit the `.spec` file, just like you would with any other RPM you were writing the specfile for yourself.

The source in the specfile should be pointing at a URL that can be fetched just like all of our other RPM-building repos do.

Reply:
Only one source in the spec file is pointing at a URL.
There are 3 other source files and small SUSE specific patch file that need to be used with it.
Seems simpler for these type of backports to just grab the SRPM package in the Makefile and unpack it there.

Reply:
Usually only (the) one (main) source does point at a URL.  The others are all probably small ancillary files such as scripts, patches, etc.  It's customary to just commit those to the SCM.

Quite often in backporting, the specfile needs modification anyway to work on the older OS release, so the usual method is to just unpack the spec and commit it and the ancillary files to the SCM rather than trying to have the Makefile work with the SRPM directly.

I would tend to think that this is most often the case, so trying to handle the small number of cases where it's not, I don't think is worth the acrobatics of trying to make it work.

It's also much nicer to see the specfile in the SCM rather than having to fetch an SRPM to unpack it if one wants to see it.

Reply:
Added the contents of the src.rpm to this project, except for the tarball.
Added a source; target to fetch the source as the suse 'build' utility expects it to be there.